### PR TITLE
(CDAP-2899) Make MR job runner in local mode use a per run directory …

### PR DESCRIPTION
**Pull request by chtyim**
…so that it gets cleanup properly after job is done.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java [hunk 1]
```diff
@@ -177,32 +177,30 @@ protected String getServiceName() {
...
-    Configuration mapredConf = job.getConfiguration();
+    // Creates a temporary directory locally for storing all generated files.
+    File tempDir = createTempDirectory();
+    this.cleanupTask = createCleanupTask(tempDir);
 
-    classLoader = new MapReduceClassLoader(context.getProgram().getClassLoader());
-    mapredConf.setClassLoader(new WeakReferenceDelegatorClassLoader(classLoader));
-    ClassLoaders.setContextClassLoader(mapredConf.getClassLoader());
+    try {
+      Job job = createJob(new File(tempDir, "mapreduce"));
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java]: ``programType`` is already part of the ``tempDir``, do we need a subdirectory called "mapreduce" too?

Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java]: Yes. This is to separate out the directory used by CDAP vs directory used by the MapReduce framework.

Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java]: :+1:

**Comment by bdmogal** [comment]
LGTM! Thanks @chtyim !

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-RBT356

**Pull request merged**